rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -73,18 +76,48 @@ def _create_states(
         Returns:
             a list of states
         """
-        return tracker.past_states(
-            domain,
-            omit_unset_slots=omit_unset_slots,
-            ignore_rule_only_turns=ignore_rule_only_turns,
-            rule_only_data=rule_only_data,
+        return cls._remove_action_unlikely_intent(
```

Comment:
Don't think this is correct. As mentioned in the [issue description](https://github.com/RasaHQ/rasa/issues/8437), the decision to remove `action_unlikely_intent` from the states depends on which policy is requesting the features for the tracker. Hence, it should only be removed for all policies which use story data for training (`TEDPolicy` and `IntentTEDPolicy`), but shouldn't be removed for others like `RulePolicy`. It's similar to how rule turns are hidden only for the former policies and not latter. See [line](https://github.com/RasaHQ/rasa/blob/3d68c685c92bb423aebaf521987990c2d89a1a35/rasa/core/policies/policy.py#L263). Since you have added the call to `_remove_action_unlikely_intent` inside `_create_states` function of the parent class, it will always be removed.

Reply:
In the latest commit, I have removed `action_unlikely_intent` filtering from `_create_states` and put it in the `FullDialogue/MaxHistory/IntentMaxHisory-TrackerFeaturizer` implementations of `prediction_states` and `training_states_labels_and_entities`.